src/nmsgtool.c

```diff
@@ -411,6 +448,47 @@ setup_nmsg_input(nmsgtool_ctx *c, nmsg_input_t input) {
 
 /* Private functions. */
 
+#ifdef HAVE_PROMETHEUS
+
+static void
+init_prometheus_counters(void)
+{
+	const char *label = "nmsgtool";
+
+	/* NMSG payload counters */
+	INIT_PROM_CTR_L(total_payloads_in, "total_payloads_in", "total number of nmsg payloads received", label);
+	assert(total_payloads_in != NULL);
+	INIT_PROM_CTR_L(total_payloads_out, "total_payloads_out", "total number of nmsg payloads sent", label);
+	assert(total_payloads_out != NULL);
+
+	/* NMSG container counters */
+	INIT_PROM_CTR_L(total_container_recvs, "total_container_recvs", "total number of nmsg containers received", label);
+	assert(total_container_recvs != NULL);
+	INIT_PROM_CTR_L(total_container_drops, "total_container_drops", "total number of nmsg containers lost", label);
+	assert(total_container_drops != NULL);
+}
+
+static int nmsgtool_prom_handler(void *clos) {
+	const char *label = "nmsgtool";
+	nmsg_io_t io = (nmsg_io_t) clos;
+	uint64_t last_sum_in = 0, last_sum_out = 0, last_container_drops = 0, last_container_recvs = 0;
+	uint64_t sum_in = 0, sum_out = 0, container_drops = 0, container_recvs = 0;
+	nmsg_io_get_stats(io, &sum_in, &sum_out, &container_recvs, &container_drops);
+
+	prom_counter_add(total_payloads_in, sum_in - last_sum_in, &label);
```

Comment:
Might be good to set return value to -1 if any call to `prom_counter_add()` fails